pyomo/contrib/gdpopt/nlp_solve.py

```diff
@@ -311,19 +312,17 @@ def detect_unfixed_discrete_vars(model):
 
 def preprocess_subproblem(m, config):
     """Applies preprocessing transformations to the model."""
-    # fbbt(m, integer_tol=config.integer_tolerance)
+    # First do FBBT
+    fbbt(m, integer_tol=config.integer_tolerance,
+         feasibility_tol=config.constraint_tolerance)
     xfrm = TransformationFactory
-    xfrm('contrib.propagate_eq_var_bounds').apply_to(m)
+    # Now that we've tightened bounds, see if any variables are fixed because
+    # their lb is equal to the ub (within tolerance)
     xfrm('contrib.detect_fixed_vars').apply_to(
-        m, tolerance=config.variable_tolerance)
-    xfrm('contrib.propagate_fixed_vars').apply_to(m)
+         m, tolerance=config.variable_tolerance)
+    # Now, if something got fixed to 0, we might have 0*var terms to remove
     xfrm('contrib.remove_zero_terms').apply_to(m)
```

Comment:
It does. I think that this transformation is called because `deactivate_trivial_constraints` decides something might be trivial by checking if the polynomial degree of the constraint body is 0. And if you have
```
m.x = Var()
m.y = Var()
m.y.fix(0)
m.e = Expression(expr=m.x*m.y)
```
then `m.e.polynomial_degree()` is 1 apparently.

So I think that the long-term solution is a rewrite of `deactivate_trivial_constraints` (I'll submit an issue so I don't forget), but for now I guess this makes sense.

Reply:
Got it!